web/src/components/SignUpDialogs.tsx

```diff
@@ -0,0 +1,225 @@
+import { Content } from "@/components/Content";
+import { AlertVariant } from "@/components/njwds-extended/Alert";
+import { Button } from "@/components/njwds-extended/Button";
+import { ToastAlert } from "@/components/njwds-extended/ToastAlert";
+import { Icon } from "@/components/njwds/Icon";
+import { IsAuthenticated } from "@/lib/auth/AuthContext";
+import { triggerSignIn } from "@/lib/auth/sessionHelper";
+import { onSelfRegister } from "@/lib/auth/signinHelper";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { useMountEffectWhenDefined } from "@/lib/utils/helpers";
+import { AuthAlertContext } from "@/pages/_app";
+import Config from "@businessnjgovnavigator/content/fieldConfig/config.json";
+import { RegistrationStatus } from "@businessnjgovnavigator/shared";
+import { Box, Dialog, DialogContent, DialogTitle, IconButton } from "@mui/material";
+import { useRouter } from "next/router";
+import React, { ReactElement, ReactNode, useContext, useEffect } from "react";
+
+export const SelfRegToast = (): ReactElement => {
+  const { isAuthenticated, registrationAlertStatus, setRegistrationAlertStatus } =
+    useContext(AuthAlertContext);
+
+  useEffect(() => {
+    if (registrationAlertStatus == "IN_PROGRESS" && isAuthenticated == IsAuthenticated.TRUE) {
+      setRegistrationAlertStatus("SUCCESS");
+    }
+  }, [isAuthenticated, registrationAlertStatus, setRegistrationAlertStatus]);
+
+  const alertMap: Record<Exclude<RegistrationStatus, "IN_PROGRESS">, AlertVariant> = {
+    SUCCESS: "success",
+    DUPLICATE_ERROR: "error",
+    RESPONSE_ERROR: "error",
+  };
+
+  const contentMap: Record<Exclude<RegistrationStatus, "IN_PROGRESS">, string> = {
```

Comment:
we could define `Exclude<RegistrationStatus, "IN_PROGRESS">` as its own type in this file